system/Common.php

```diff
@@ -270,8 +270,16 @@ function esc($data, $context = 'html', $encoding = null)
 				$method = 'escape' . ucfirst($context);
 			}
 
-			// @todo Optimize this to only load a single instance during page request.
-			$escaper = new \Zend\Escaper\Escaper($encoding);
+                       static $escaper;
+                       if (! $escaper)
+                       {
+			        $escaper = new \Zend\Escaper\Escaper($encoding);
+                       }
+
+                       if ($encoding && $escaper->getEncoding() !== $encoding)
+                       {
+                               $escaper = new \Zend\Escaper\Escaper($encoding);
+                       }
```

Comment:
I've updated, when `$encoding` is null, actually, `Zend\Escaper` uses 'utf-8', when user passed non-null of encoding, I added check with `$escaper->getEncoding() !== $encoding`.